### PR TITLE
feat(code-suggester)!: move to code-suggester for PR management

### PR DESCRIPTION
**Pull request by bcoe**
integrate release-please with code-suggester, benefits:

1. the release PR comes in as a single commit (allowing us to stop adding `[skip ci]`).
2. it seems as though this stops the behavior of PRs opening and closing as they're updated.

Depends on: https://github.com/googleapis/code-suggester/pull/95

### CODE REVIEW

**Comment by chingor13** [comment]
This looks nice!

**Review by JustinBeckwith — APPROVED** [review]
LGTM with annoying nits

**Review by alexander-fenster — APPROVED** [review]

File: src/github.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import {createPullRequest} from 'code-suggester';
+import {Changes} from 'code-suggester/build/src/types/index';
```
[the remark this answers is not in the document]
Comment (bcoe) [reply src/github.ts:16]: done: https://github.com/googleapis/code-suggester/pull/98

File: src/github.ts [hunk 2]
```diff
@@ -596,97 +602,43 @@ export class GitHub {
...
-          // that the user does not have access to the repo:
-          throw new AuthError();
-        } else {
-          throw err;
-        }
+    // check if there's an existing PR, so that we can opt to update it
+    // rather than creating a new PR.
+    const refName = `refs/heads/${options.branch}`;
+    let openReleasePR: PullsListResponseItem | undefined;
+    (await this.findOpenReleasePRs(options.labels)).forEach(releasePR => {
```
[the remark this answers is not in the document]
Comment (bcoe) [reply src/github.ts:611]: have moved to two lines in my working branch.

File: src/github.ts [hunk 3]
```diff
@@ -596,97 +602,43 @@ export class GitHub {
...
+      if (refName && refName.includes(releasePR.head.ref)) {
```
[the remark this answers is not in the document]
Comment (bcoe) [reply src/github.ts:612]: switched to `for`/`of` with `break`.

File: src/github.ts [hunk 4]
```diff
@@ -704,30 +656,65 @@ export class GitHub {
...
+    }
+  }
+
+  private async getChangeSet(
+    updates: Update[],
+    defaultBranch: string
+  ): Promise<Changes> {
+    const refName = `refs/heads/${defaultBranch}`;
+    const changes = new Map();
+    for (let i = 0; i < updates.length; i++) {
```
[the remark this answers is not in the document]
Comment (bcoe) [reply src/github.ts:673]: Switched to:

`for (const update of updates)`

I think the `i` was used in a previous iteration of the function.

File: src/github.ts [hunk 5]
```diff
@@ -596,97 +602,43 @@ export class GitHub {
...
-        } else {
-          throw err;
-        }
+    // check if there's an existing PR, so that we can opt to update it
+    // rather than creating a new PR.
+    const refName = `refs/heads/${options.branch}`;
+    let openReleasePR: PullsListResponseItem | undefined;
+    (await this.findOpenReleasePRs(options.labels)).forEach(releasePR => {
+      if (refName && refName.includes(releasePR.head.ref)) {
+        openReleasePR = releasePR as PullsListResponseItem;
```
[the remark this answers is not in the document]
Comment (bcoe) [reply src/github.ts:613]: I now `break`.

**Pull request merged**